chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
> Even if we save previous_epoch_detail in triggers, the issue, how to treat the cases that updaters/triggers were serialized by older version and do not have previous_epoch_detail in it

Sorry, my reply was about the difficulty to use `Updater.previous_epoch_detail` and did not fully answer to your original question.

As `IntervalTrigger` cannot rely on `Updater.previous_epoch_detail`, it should store and serialize the previous value of `epoch_detail` (as e.g. `IntervalTrigger.previous_epoch_detail`).  When `previous_epoch_detail` attribute is missing in deserializing `IntervalTrigger`, there are two choices:
1. Do not rely on `Updater.previous_epoch_detail`. Set some special value that indicates (e.g. `None`, `-1`) and to `IntervalTrigger.previous_epoch_detail` and raise a warning that notifies users that there could be a resuming problem.
2. Try to retrieve the value `Updater.previous_epoch_detail`, assuming that the trigger is called at every iteration. If it is missing, raise a warning of the resuming problem.

If there is a way to infer the previous value of `epoch_detail` from trainer, 
It can take invalid values for indicating the first iteration (e.g `None` or `-1`).

1.  Try to get the value from `IntervalTrigger.previous_epoch_detail`.
2. If `IntervalTrigger.previous_epoch_detail` is missing, try to get the value from `Updater.previous_epoch_detail`, assuming that the trigger is called at every iteration. If it is successful, we store the value to `IntervalTrigger.previous_epoch_detail`.
3. If `Updater.previous_epoch_detail`, we set previous_epoch_detail to